Source/Foundation/bsfCore/Renderer/BsCamera.cpp

```diff
@@ -14,6 +14,9 @@
 #include "Renderer/BsRenderer.h"
 #include "Scene/BsSceneManager.h"
 
+// Do not delete this, it prevents massive problems with windows headers across the codebase
+#include "Private/Win32/BsWin32Undefs.h"
```

Comment:
Having a private header in a public header will not work since private headers don't get installed, so it will break in binary distributions. Also, I'd prefer to move this to a more generic location - perhaps even replacing all usages of "windows.h" with a custom header that automatically undefines the problematic macros?

Reply:
Isn't this a public source file anyway?

Either way, I agree that replacing windows.h with a proxy header would be better. I'll get on that.

Reply:
Ah sorry, I thought this was `BsCamera.h`, my bad.

Reply:
Cotire only walks source files to build the PCH list, (though it gets transitively included ones) so the best way (besides doing what you suggested) is to have it included directly in a source file.